### DISCUSSION TITLE
current state of an entity - anyone know how I can get all the states of a device?

### DISCUSSION THREAD

**Discussion by disruptivepatternmaterial** [zachowj/node-red-contrib-home-assistant-websocket#1322] [Q&A]
I am scratching my head that I cannot find someone who has done this, so I am guessing I am asking the wrong question 😃 !

What I want is to get all the values (entities) for an entire device. Why? I have a TON of devices, sensors, etc... that I feed into influx, but I really dislike the HA built in one. So I want to periodically query the values of sensors and such to then feed grafana dashboards. Why? I have lots of things that do not run through HA, so I have my own schema...

But here is the thing - if something is not changing often, I cannot tell if it is not changing or just dead. It makes sense to just asking for a status every 5 min - even if every 5 min the number is unchanged.

Ideas?

**Comment by zachowj** [comment]
Use a current state node. You can put any valid entity in the entity id field.

In the output properties set a message property to `$deviceEntities("devceID")` with the type set to J: expression and replace `deviceID` with a valid device id.

<https://zachowj.github.io/node-red-contrib-home-assistant-websocket/guide/jsonata.html>

**Reply by disruptivepatternmaterial** [reply]
I appreciate the reply, however that doesnt seem to have the effect I mentioned above.

I want all the entities for a device without having to create 500 current state nodes.

The ask is if there is something like a current state that takes a device as input instead of an entity.

**Reply by zachowj** [reply]
You can change JSONata so that the device id is dynamic from the message object.

```
$deviceEntities(payload)
```

**Reply by disruptivepatternmaterial** [reply]
Again appreciate the reply, but something is not lining up here.

The current state takes in an entity id, if I put in $deviceEntities(payload) and pass in a device or entity (I tried both) in the payload using an inject all I get are errors.

The goal is to periodically query a device and have a structured reply of something like { entity1: 1, entity2: 5, ... }

Basically I would like the poll or state nodes to take wildcards, maybe that helps explain.
